ddht/tools/web3.py

```diff
@@ -0,0 +1,85 @@
+from typing import Callable, Mapping, NamedTuple, Tuple
+
+try:
+    import web3  # noqa: F401
+except ImportError:
+    raise ImportError("The web3.py library is required")
+
+
+from eth_enr import ENR, ENRAPI
+from eth_typing import NodeID
+from eth_utils import decode_hex
+from web3.method import Method
+from web3.module import ModuleV2
+from web3.types import RPCEndpoint
+
+from ddht.rpc_handlers import BucketInfo as BucketInfoDict
+from ddht.rpc_handlers import NodeInfoResponse, TableInfoResponse
+
+
+class NodeInfo(NamedTuple):
```

Comment:
No strong preference, but in the interests of removing duplication it might be better to add these `from_rpc_response` methods to the original classes back in `rpc_handlers`. These classes are duplicates which must always be kept in sync with the originals!

Reply:
Though, wouldn't it be cool if there was a json library which took a `NamedTuple` like this and populated it from the received json object, throwing errors for missing or mistyped fields. If these classes were any more complicated I'd look into using [marshmallow](https://marshmallow.readthedocs.io/en/stable/).

Reply:
They are implemented here so that the web3 specific stuff doesn't leak into the other RPC stuff.  Yes they have to be kept inline, but they are also all strongly typed and mypy tells us right away when they diverge (in most cases....).  Unless you feel strongly.  Also, `TypedDict` doesn't allow you to add methods.  It is pure typing sugar.